test/distributed/pipeline/sync/test_pipe.py

```diff
@@ -446,23 +455,26 @@ def test_deferred_batch_norm_params(checkpoint, setup_rpc):
     assert torch.allclose(pipe[0].bias.grad, bn.bias.grad, atol=1e-4)
 
 
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="cuda required")
 def test_devices(setup_rpc):
     a = nn.Linear(1, 1)
-    b = nn.Linear(1, 1)
+    b = nn.Linear(1, 1).to(0)
     c = nn.Linear(1, 1)
 
     # There are extra two devices.
     model = nn.Sequential(a, b, c)
     model = Pipe(model)
 
     cpu = torch.device("cpu")
+    cuda_0 = torch.device(0)
     # Extra devices must be discarded.
-    assert model.devices == [cpu, cpu, cpu]
+    assert model.devices == [cpu, cuda_0, cpu]
 
 
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="cuda required")
```

Comment:
Shouldn't we also add a new test to validate the new functionality, I'm not sure if these updates to existing tests cover this.

Reply:
good point, added a test